File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        int[] num = new int[10000001];
        
        for(int i =0;i<tangerine.length;i++){
            int index = tangerine[i];
            num[index]++;
        }
        
        Arrays.sort(num);
        
        for(int i=num.length-1;i>=0;i--){
            k -= num[i];
            answer++;
            if(k <=0) break;
        }
        
        
        
        
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <bits/stdc++.h>

using namespace std;

bool cmp(pair<int,int> &a, pair<int,int> &b) {
    if (a.second == b.second) return a.first < b.first;
    return a.second > b.second;
}

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    int cnt = 0;
    
    map<int,int> m;
    
    for (auto &i : tangerine) {
        m[i]++;
    }
    
    vector<pair<int,int>> v(m.begin(), m.end());
    sort(v.begin(), v.end(), cmp);
    
    for (int i=0;i<v.size();i++) {
        answer++;
        
        cnt += v[i].second;
        if (cnt >= k) break;
    }
    
    
    return answer;
}